### PR TITLE
Add a default task parser

### LINKED ISSUE #153: Make a default task parser
Currently, all reviewbot tasks are parsed based their [suite type](https://github.com/mozilla/code-review/blob/master/bot/code_review_bot/workflow.py#L278-L302). Given that a default analysis format [exists](https://github.com/mozilla/code-review/blob/master/docs/analysis_format.md), we should have a default task parser as well (similar to `MozLintTask`).

Without this, the mozlint test type definitions (in taskcluster) will start to get messy since we have to add non-linting related tasks to it (or the other existing source-tests) to get them to run in reviewbot.

### PR DESCRIPTION
**Pull request by La0**
Fixes #153

@sylvestre if you can take a look at [the updated format in the docs](https://github.com/La0/code-review/blob/165f8203a2bf20d5ed44ad4f0e9387c07eea9051/docs/analysis_format.md)

### CODE REVIEW

File: docs/analysis_format.md [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
 
-The [default format](https://github.com/mozilla/code-review/blob/1.0.2/bot/code_review_bot/tasks/lint.py#L170) (loosely based on Mozlint format) has the following fields for each issue:
+The file **must** be available as `public/code-review/issues.json`.
 
-* `path` relative to the repository
-* `column` & `lineno` where the issue is happening in the file
+The [default format](https://github.com/mozilla/code-review/blob/1.0.5/bot/code_review_bot/tasks/default.py#L170) (loosely based on Mozlint format) has the following fields for each issue:
+
+* `path` **relative** to the repository
+* `column` & `line` where the issue is happening in the file. They must be positive integers, or `null` when unknown or for a full file.
```
Comment (marco-c) [inline docs/analysis_format.md]:
```suggestion
* `column` & `line` where the issue is happening in the file. They must be positive integers, or `null` when unknown or for an issue linked to a full file.
```

File: bot/code_review_bot/tasks/default.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        return [
+            DefaultIssue(
+                analyzer=issue.get("analyzer", self.name),
+                revision=revision,
+                path=issue["path"],
+                line=issue["line"],
+                column=issue["column"],
+                nb_lines=issue.get("nb_lines", 1),
+                level=issue["level"],
+                check=issue["check"],
```
Comment (marco-c) [inline bot/code_review_bot/tasks/default.py]: Maybe we should default the check to the analyzer name like we do for MozLint (I guess many analyzers will be single-rule analyzers).

File: bot/code_review_bot/workflow.py [hunk 3]
```diff
@@ -294,9 +295,15 @@ def build_task(self, task_id, task_status):
...
         elif name == "source-test-infer-infer":
             return InferTask(task_id, task_status)
-        elif name.startswith("source-test-"):
-            logger.error(
-                "Unsupported source-test task: will need a local implementation"
-            )
         else:
-            raise Exception("Unsupported task {}".format(name))
+
+            # If default artifacts available > boom
```
Comment (marco-c) [inline bot/code_review_bot/workflow.py]: We should file an issue (if we don't already have one) to move all analyzers to the common format, so we can cleanup a bit the code.

File: bot/code_review_bot/tasks/default.py (hunk 2 above)
Comment (marco-c) [reply bot/code_review_bot/tasks/default.py]: If you do this, don't forget to update the docs too

File: bot/code_review_bot/workflow.py (hunk 3 above)
Comment (La0) [reply bot/code_review_bot/workflow.py]: It probably would be better to file a Bugzilla meta bug ?

File: bot/code_review_bot/tasks/default.py (hunk 2 above)
Comment (La0) [reply bot/code_review_bot/tasks/default.py]: Done

File: bot/code_review_bot/workflow.py (hunk 3 above)
Comment (marco-c) [reply bot/code_review_bot/workflow.py]: Yep! And an issue in this repo to get rid of the code when that lands in mozilla-central

Comment (La0) [reply bot/code_review_bot/workflow.py]: Filed [Bug 1590669](https://bugzilla.mozilla.org/show_bug.cgi?id=1590669) & #180

**Pull request merged**